sympy/sets/dispatchers.py

```diff
@@ -150,6 +241,18 @@ def function_sets(f, x):
         return Interval(log(x.start), log(x.end), x.left_open, x.right_open)
     return ImageSet(Lambda(_x, f(_x)), x)
 
-@dispatch(FunctionClass, Set)
+@dispatch(FunctionUnion, Union)
+def function_sets(f, x):
+    return Union(imageset(f, arg) for arg in x.args)
+
+@dispatch(FunctionUnion, Intersection)
+def function_sets(f, x):
+    return Intersection(imageset(f, arg) for arg in x.args)
```

Comment:
Intersection of image sets may contain elements that are not in the image of the intersection. Compare the intersection of  `f([-1, 0])` and `f([0, 1])` with the image of the intersection `f({0})` when `f(x) = x**2`.

Reply:
I see. This code was just copied from the `_eval_imageset` method of Intersection. It is supposed to be called when the intersection cannot be calculated.

Shall we leave the imageset unevaluated for this case?

Reply:
Bijective functions may be treated separately.

Reply:
> Shall we leave the imageset unevaluated for this case?

I guess that is what has to be done.